libs/providers/flagd/src/lib/flagd-provider.ts

```diff
@@ -53,19 +43,19 @@ export class FlagdProvider implements Provider {
         : new GRPCService(config, undefined, logger);
   }
 
-  initialize(): Promise<void> {
-    return this._service
-      .connect(this.handleReconnect.bind(this), this.handleChanged.bind(this), this.handleError.bind(this))
-      .then(() => {
-        this.logger?.debug(`${this.metadata.name}: ready`);
-        this._status = ProviderStatus.READY;
-      })
-      .catch((err) => {
-        this._status = ProviderStatus.ERROR;
-        this.logger?.error(`${this.metadata.name}: error during initialization: ${err.message}`);
-        this.logger?.debug(err);
-        throw err;
-      });
+  async initialize(): Promise<void> {
+    try {
+      await this._service.connect(
+        this.handleReconnect.bind(this),
+        this.handleChanged.bind(this),
+        this.handleError.bind(this),
+      );
+      this.logger?.debug(`${this.metadata.name}: ready`);
+    } catch (err) {
+      this.logger?.error(`${this.metadata.name}: error during initialization: ${(err as Error)?.message}`);
```

Comment:
Do we want to leave this as log level error?

Reply:
I _think_ so, for debugging init (port numbers wrong, etc).